rio_cogeo/cogeo.py

```diff
@@ -446,7 +453,10 @@ def cog_validate(src_path: str, strict: bool = False, quiet: bool = False):
         for e in errors:
             click.echo("- " + e, err=True)
 
+    is_valid = True
     if errors or (warnings and strict):
-        return False
+        is_valid = False
 
-    return True
+    if return_errors:
+        return {"errors": errors, "warnings": warnings, "valid": is_valid}
+    return is_valid
```

Comment:
It feels clunky to have two very different return types

Reply:
Agreed, I was trying to avoid a breaking change but maybe it can be breaking since this would be a beta release

Reply:
>  maybe it can be breaking since this would be a beta release

yep go ahead

Reply:
should we return a `Tuple(is_valid,  errors, warnings)` ?

Reply:
Done in https://github.com/cogeotiff/rio-cogeo/pull/143/commits/7738517290fd3c0748b45b67594dfc2efa7c3168